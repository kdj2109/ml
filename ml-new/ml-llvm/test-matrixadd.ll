; ModuleID = 'ML'

@fmt = private unnamed_addr constant [4 x i8] c"%d\0A\00"
@fmt1 = private unnamed_addr constant [4 x i8] c"%f\0A\00"
@fmt2 = private unnamed_addr constant [4 x i8] c"%d\0A\00"
@fmt3 = private unnamed_addr constant [4 x i8] c"%f\0A\00"

declare i32 @printf(i8*, ...)

define i32 @main() {
entry:
  %a = alloca [3 x [4 x i32]]
  %b = alloca [2 x [2 x i32]]
  %c = alloca [3 x i32]
  %i = alloca i32
  store [3 x [4 x i32]] [[4 x i32] [i32 1, i32 2, i32 3, i32 4], [4 x i32] [i32 3, i32 4, i32 5, i32 6], [4 x i32] [i32 7, i32 8, i32 9, i32 8]], [3 x [4 x i32]]* %a
  store [2 x [2 x i32]] [[2 x i32] [i32 5, i32 7], [2 x i32] [i32 9, i32 15]], [2 x [2 x i32]]* %b
  %a1 = getelementptr inbounds [3 x [4 x i32]]* %a, i32 0, i32 0, i32 0
  ret i32 0
}

define i32** @addTuples2(i32** %z, i32** %y) {
entry:
  %z1 = alloca i32**
  store i32** %z, i32*** %z1
  %y2 = alloca i32**
  store i32** %y, i32*** %y2
  %z3 = load i32*** %z1
  ret i32** %z3
}
