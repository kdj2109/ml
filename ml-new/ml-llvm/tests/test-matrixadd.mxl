void addIntMatrices(int[][] x, int[][] y, int rows, int columns) {
	int len;
	int i;

	len = rows * columns;

	for(i = 0; i < len; i = i + 1) {
		$x = $x + $y;
		x = x.+;
		y = y.+;
	}
}

int main() {
	int[3:4] a;
	int[3:4] b;
	int i;
	int j;

	a = [| 1, 2, 3, 4 | 3, 4, 5, 6 | 7, 8, 9, 8 |];
	b = [| 5, 7, 3, 4 | 9, 15, 7, 8 | 1, 2, 3, 4 |];


	addIntMatrices(@@a, @@b, 3, 4);

	for(i = 0; i < 3; i = i + 1) {
		for (j = 0; j < 4; j = j + 1) {
			print(a[i:j]);
		}
	}

	return 0;
}
