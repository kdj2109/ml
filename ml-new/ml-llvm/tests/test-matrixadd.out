6
9
6
8
12
19
12
14
8
10
12
12
